app/models/event.rb

```diff
@@ -39,6 +39,7 @@ class Event < ActiveRecord::Base
 
   scope :confirmed, -> { where(state: 'confirmed') }
   scope :highlighted, -> { where(is_highlight: true) }
+  scope :schedule, ->(schedule_id) {  event_schedules.find_by(schedule_id: (schedule_id.present? ? schedule_id : program.try(:selected_schedule))) }
```

Comment:
This is wrong. I think it can not be an scope, because scopes are for calls on models or association object, but the event_schedules is of an specific event. I'll move that to a private function in Event.